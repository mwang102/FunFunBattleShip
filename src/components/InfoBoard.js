import React from 'react'
import styled from "styled-components";
import PropTypes from 'prop-types'

const TableData = styled.td`
	font-size:2rem;
    border: 1px solid #dddddd;
    text-align: left;
    padding: 8px;
`

const TableContainer = styled.table`
	flex:1;
	margin:1rem;
    white-space: nowrap;

`

export default class InfoBoard extends React.Component{
	static propTypes = {
		turn:PropTypes.number.isRequired,
		hits:PropTypes.number.isRequired,
		miss:PropTypes.number.isRequired,
		sunkBoats:PropTypes.number.isRequired,
		alreadyHit:PropTypes.number.isRequired,
		numberHitsLeft:PropTypes.number.isRequired,
	}

 	render(){
 		const { 
 			turn,
 			hits,
 			miss,
 			sunkBoats,
 			alreadyHit,
 			numberHitsLeft,
 		} = this.props

 		return(
			 <TableContainer>
				<tbody>
				  <tr>
				    <TableData>Turns:</TableData>
				    <TableData>{turn}</TableData>
				  </tr>
				  <tr>
				    <TableData>Hits:</TableData>
				    <TableData>{hits}</TableData>
				  </tr>
				  <tr>
				    <TableData>Miss:</TableData>
				    <TableData>{miss}</TableData>
				  </tr>
				  <tr>
				    <TableData># Sunk:</TableData>
				    <TableData>{sunkBoats}</TableData>
				  </tr>
				  <tr>
				    <TableData>Already Hit :</TableData>
				    <TableData>{alreadyHit}</TableData>
				  </tr>
				  <tr>
				    <TableData>Number Hits Left :</TableData>
				    <TableData>{numberHitsLeft}</TableData>
				  </tr>
			  </tbody>
			</TableContainer>
 			)
 	}
}