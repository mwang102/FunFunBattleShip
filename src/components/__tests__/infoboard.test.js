import React from 'react';
import { shallow } from 'enzyme'
import InfoBoard from '../InfoBoard'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import styled from "styled-components";

describe('Info Board', ()=>{
	it('Info props are matched to correct location', ()=>{ 
		const wrapper = shallow(<InfoBoard numberHitsLeft={5} 
										   alreadyHit={0} 
										   turn={0} 
										   hits={0} 
										   miss={0} 
										   sunkBoats={0}/>)

		const infoBoardTableTurnData = wrapper.find('InfoBoard__TableData').at(1).render().text()
		expect(infoBoardTableTurnData).toBe('0')

		const infoBoardTableHitData = wrapper.find('InfoBoard__TableData').at(3).render().text()
		expect(infoBoardTableHitData).toBe('0')

		const infoBoardTableMissData = wrapper.find('InfoBoard__TableData').at(5).render().text()
		expect(infoBoardTableMissData).toBe('0')

		const infoBoardTableNumberSunk = wrapper.find('InfoBoard__TableData').at(7).render().text()
		expect(infoBoardTableNumberSunk).toBe('0')

		const infoBoardTableAlreadyHit = wrapper.find('InfoBoard__TableData').at(9).render().text()
		expect(infoBoardTableAlreadyHit).toBe('0')

		const infoBoardTableNumberHitsLeft = wrapper.find('InfoBoard__TableData').at(11).render().text()
		expect(infoBoardTableNumberHitsLeft).toBe('5')
	
	})


	it('matches the snapshot', ()=>{
		const tree = renderer.create(<InfoBoard numberHitsLeft={5} 
										   alreadyHit={0} 
										   turn={0} 
										   hits={0} 
										   miss={0} 
										   sunkBoats={0}/>).toJSON()
		
		expect(tree).toMatchSnapshot()
	}) 

})