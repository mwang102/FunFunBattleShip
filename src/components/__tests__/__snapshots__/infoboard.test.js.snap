// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Board matches the snapshot 1`] = `
.c1 {
  font-size: 2rem;
  border: 1px solid #dddddd;
  text-align: left;
  padding: 8px;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 1rem;
  white-space: nowrap;
}

<table
  className="c0"
>
  <tbody>
    <tr>
      <td
        className="c1"
      >
        Turns:
      </td>
      <td
        className="c1"
      >
        0
      </td>
    </tr>
    <tr>
      <td
        className="c1"
      >
        Hits:
      </td>
      <td
        className="c1"
      >
        0
      </td>
    </tr>
    <tr>
      <td
        className="c1"
      >
        Miss:
      </td>
      <td
        className="c1"
      >
        0
      </td>
    </tr>
    <tr>
      <td
        className="c1"
      >
        # Sunk:
      </td>
      <td
        className="c1"
      >
        0
      </td>
    </tr>
    <tr>
      <td
        className="c1"
      >
        Already Hit :
      </td>
      <td
        className="c1"
      >
        0
      </td>
    </tr>
    <tr>
      <td
        className="c1"
      >
        Number Hits Left :
      </td>
      <td
        className="c1"
      >
        5
      </td>
    </tr>
  </tbody>
</table>
`;
